packages/jaeger-ui/src/components/DeepDependencyGraph/Graph.tsx

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { Component } from 'react';
+import { DirectedGraph, LayoutManager } from '@jaegertracing/plexus';
+import _map from 'lodash/map';
+import { History as RouterHistory, Location } from 'history';
+import queryString from 'query-string';
+
+import GraphModel from '../../model/ddg/Graph';
+import { encode } from '../../model/ddg/visibility-codec';
+
+import { PathElem, TDdgModel } from '../../model/ddg/types';
+
+type TProps = {
+  history: RouterHistory;
+  location: Location;
+  ddgModel: TDdgModel;
+  visKey?: string;
+};
+
+export default class Graph extends Component<TProps> {
+  private graphModel: GraphModel;
+  private layoutManager: LayoutManager;
+
+  constructor(props: TProps) {
+    super(props);
+    const { ddgModel } = props;
+    const { distanceToPathElems } = ddgModel;
+    this.graphModel = new GraphModel({ ddgModel });
+    this.layoutManager = new LayoutManager({ useDotEdges: true, splines: 'polyline' });
+
+    // TODO: Discuss if this should be in a componentDidUpdate in case visKey is unset without this component
+    // being re-mounted
+    if (!this.props.visKey) {
```

Comment:
this can be used in the render function without modifying the url

Reply:
should be a utility `getDefaultVisEncoding`

Reply:
prop should be `visEncoding` throughout

Reply:
or `Graph.getVisible` should take `visEncoding` as an optional argument, and default to the two hop behavior.
likewise for the future `Graph.showChildren(vertexKey: string, currVisEncoding?: string)`, `Graph.hideChildren(vertexKey: string, currVisEncoding?: string)`, `Graph.showHop(hop: number /* positive or negative */, currVisEncoding?: string)`